docker/db-migration/migrations/root.yaml

```diff
@@ -0,0 +1,8 @@
+---
+databaseChangeLog:
+  - changeSet:
+      id: root-tag
+      author: surabujin
+      changes:
+        - tagDatabase:
+            tag: history-starting-point
```

Comment:
I was thinking about tags as a solution for automatic rollback procedure, not manual.
Tags can be aligned to docker images, so the database update procedure  performs upgrade/downgrade in one action using the same docker image. 

Here's the scenario how the migration pipeline may look like:
- db-migration images for RC1 and RC2 are built using the git commit hash as the tag.
- During RC1 deployment, the database is upgraded and tagged with the RC1 hash.
- During RC2 deployment, the database is upgraded and tagged with the RC2 hash.
- During RC1 deployment, the database is downgraded with rolling back to the RC1 hash.

Reply:
It is good idea, but there is no simple way to implement it. There is at least 2 big challenge:
* to perform rollback we need migrations set used during corresponding update, because rollback instructions stored inside migration's set (i.e if we make update using hash H1 we need same container/code with hash H1 to do rollback)
* we need to find a common root for releases to make rollback (i.e. we have deployed RC1 and we are going to deploy RC2, we don't know how many tags we should rollback in RC1 before starting update to RC2).

Or we can choose another way for rollbacks - use rollback-sql generation during updates, and keep it as artifacts somewhere... and use them, somehow to rollback. There is a lot of challenge here too. :)

Reply:
Ok, let's continue rollback / tagging development as a separate task.